### PR TITLE
feature: added support for dynamic routes

### PR DESCRIPTION
**Pull request by mickael-btc**
I added support for dynamic routes. As before Thyra was only searching for static files.
For website using react-router, the routing is handled on the index.html file. It is now possible on Thyra.

Furthermore for simple static website that don't implement routing, there is a possibility to add a 404.html to to zip.
If a request doesn’t match any file and the user created a 404.html, Thyra will return this file with the associated status code. Otherwise Thyra will return the index.html page and it is up to the client to manage the request.

### CODE REVIEW

File: pkg/onchain/website/browser.go [hunk 1]
```diff
@@ -10,13 +10,24 @@ import (
...
-	return m[filename], nil
+	if _, ok := m[filename]; !ok {
+		// if the file is not found, try to find 404.html
+		if _, ok := m["404.html"]; ok {
+			return m["404.html"], false, 404, nil
+		}
+
+		// otherwise, let the client handle the route
+		// return true to indicate that the route is dynamic
+		return m["index.html"], true, 200, nil
```
Comment (gregLibert) [inline pkg/onchain/website/browser.go:27]: This should be defined in a web site configuration linked with the web site.

File: pkg/onchain/website/browser.go [hunk 2]
```diff
@@ -42,12 +53,21 @@ func setContentType(file string, writer http.ResponseWriter) {
...
+	if dynamic {
+		writer.Header().Set("Content-Type", "text/html")
+	} else {
+		setContentType(resource, writer)
+	}
+
+	writer.WriteHeader(status)
+
+	// add Access-Control-Allow-Origin header (for CORS)
+	writer.Header().Set("Access-Control-Allow-Origin", "*")
```
Comment (gregLibert) [inline pkg/onchain/website/browser.go:70]: This also should be defined in a website configuration.

**Review by gregLibert — COMMENTED** [review]
Very good job @mickael-btc!

@SlnPons I think we should include this in a broader discussion around website configuration/manifest.

**Comment by SlnPons** [comment]
@gregLibert @mickael-btc thank you guys. I'll organise this discussion very soon so you can both participate to it. this will help me to understand where to start with this. 
Will keep you posted.

**Comment by SlnPons** [comment]
Hey @0xMazout if I understood the problem you discussed last week with Greg (please correct me if not) you have created issues to solve the routing problem correctly and on the long term? 
If so, could you please link them in here and close this issue please?

**Comment by 0xMazout** [comment]
> Hey @0xMazout if I understood the problem you discussed last week with Greg (please correct me if not) you have created issues to solve the routing problem correctly and on the long term? If so, could you please link them in here and close this issue please?

Yes this is the issue : https://github.com/massalabs/thyra/issues/555

**Pull request closed without merging**